### PR TITLE
docs: rework icons catalog

### PR DESCRIPTION
**Pull request by layershifter**
## Previous Behavior

image

## New Behavior

image

### Changes

- Catalog itself is virtualzed, experience with filtering is more smooth
- Copy functionality
  image
- Unsized icons have bigger size and are default now
- Icons are moved to be top level in menu
  image

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/Avatar-mount_base.html">630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/Avatar-mount.html">646</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/Button-mount_base.html">304</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/Button-mount.html">294</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/Field-mount_base.html">1086</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/Field-mount.html">1139</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/FluentProvider-mount_base.html">712</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/FluentProvider-mount.html">708</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/FluentProviderWithTheme-mount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/FluentProviderWithTheme-mount.html">90</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">54</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">65</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">71</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">68</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/MakeStyles-mount_base.html">855</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/MakeStyles-mount.html">863</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/Persona-mount_base.html">1772</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/Persona-mount.html">1764</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/SpinButton-mount_base.html">1424</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30612/perf-test-react-components/SpinButton-mount.html">1400</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `70.732 kB`<br />`20.465 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `217.207 kB`<br />`61.943 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `43.219 kB`<br />`14.299 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `7.944 kB`<br />`2.588 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressBar.fixture.js'>ProgressBar</abbr> | `17.428 kB`<br />`6.898 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/fbf194baab5ca7b4b7db6017c355fb48e5489884'>fbf194baab5ca7b4b7db6017c355fb48e5489884</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: apps/public-docsite-v9/src/DocsComponents/ReactIconGrid.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  Toaster,
+  ToastTitle,
+  tokens,
+  useId,
+  useIsomorphicLayoutEffect,
+  useScrollbarWidth,
+  useToastController,
+} from '@fluentui/react-components';
+import * as React from 'react';
+import { FixedSizeGrid, type GridChildComponentProps } from 'react-window';
```
Comment (spmonahan) [inline apps/public-docsite-v9/src/DocsComponents/ReactIconGrid.stories.tsx:22]: Just curious, don't think you need to change this, but why not use the [Fluent virtualizer](https://react.fluentui.dev/?path=/docs/preview-components-virtualizer--default)?

File: apps/public-docsite-v9/src/DocsComponents/ReactIconGrid.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+const ICONS_LIST: React.FC<ReactIcons.FluentIconsProps>[] = Object.keys(ReactIcons)
+  .map(iconName => (ReactIcons as any)[iconName])
+  .filter(icon => !!icon && !!icon.displayName);
+
+const useClasses = makeStyles({
+  controls: {
+    display: 'grid',
+    gridTemplateColumns: '1fr auto',
+    ...shorthands.gap('20px'),
```
Comment (spmonahan) [inline apps/public-docsite-v9/src/DocsComponents/ReactIconGrid.stories.tsx:35]: _super nit_: use tokens

File: apps/public-docsite-v9/src/DocsComponents/ReactIconGrid.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+  },
+
+  input: {
+    width: '100%',
+  },
+
+  radioControl: {
+    ...shorthands.borderRadius(tokens.borderRadiusSmall),
+    boxShadow: tokens.shadow2,
+    ...shorthands.padding('5px', '10px'),
```
Comment (spmonahan) [inline apps/public-docsite-v9/src/DocsComponents/ReactIconGrid.stories.tsx:51]: _super nit_: use tokens

**Review by spmonahan — APPROVED** [review]

File: apps/public-docsite-v9/src/DocsComponents/ReactIconGrid.stories.tsx (hunk 1 above)
Comment (layershifter) [reply apps/public-docsite-v9/src/DocsComponents/ReactIconGrid.stories.tsx:22]: @spmonahan this example is using grid layout. AFAIK Fluent virtualizer does not support it.

**Review by khmakoto — APPROVED** [review]

**Pull request merged**